cpp/include/cudf/detail/utilities/host_vector.hpp

```diff
@@ -99,6 +99,9 @@ class rmm_host_allocator {
   /**
    * @brief Construct from a `cudf::host_async_resource_ref`
    */
+#ifdef __CUDACC__
+#pragma nv_exec_check_disable
+#endif
```

Comment:
@bdice Could you elaborate on why this is necessary? IIUC this is a host-only API, so even when compiling with nvcc, there aren’t any launch configuration checks to disable.

Reply:
Without this, I get compilation errors. I'll generate those again and share them. Sometimes we've seen that the compiler just doesn't correctly deduce execution spaces, and this is the way we work around those. Examples: https://github.com/search?q=repo%3Arapidsai%2Fcudf%20nv_exec_check_disable&type=code